Services/TimesheetsExporter.php

```diff
@@ -26,18 +26,36 @@ public function __construct(
      *
      * @return array
      */
-    public function generateData(array $criteria): array
-    {
-        $projectId = (int) ($criteria['project'] ?? 0);
-        $kind = $criteria['kind'] ?? null;
-        $dateFrom = $this->getDateTime($criteria['dateFrom'] ?? null, new Carbon('2000-01-01'));
-        $dateTo = $this->getDateTime($criteria['dateTo'] ?? null, new Carbon('2100-01-01'));
-        $userId = (int) ($criteria['userId'] ?? 0);
-        $invEmpl = filter_var($criteria['invEmpl'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $invComp = filter_var($criteria['invComp'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $paid = filter_var($criteria['paid'] ?? null, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
-        $clientId = (int) ($criteria['clientId'] ?? 0);
-
-        return $this->timesheets->getAll($projectId, $kind, $dateFrom, $dateTo, $userId, $invEmpl, $invComp, $paid, $clientId, 0);
+    public function generateData(array $criteria): array {
+      $projectId = (int) ($criteria[ 'project' ] ?? 0);
+      $kind = $criteria[ 'kind' ] ?? NULL;
+      $dateFrom = $this->getDateTime($criteria[ 'dateFrom' ] ?? NULL, new Carbon('2000-01-01'));
+      $dateTo = $this->getDateTime($criteria[ 'dateTo' ] ?? NULL, new Carbon('2100-01-01'));
+      $userId = (int) ($criteria[ 'userId' ] ?? 0);
+      $invEmpl = filter_var($criteria[ 'invEmpl' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $invComp = filter_var($criteria[ 'invComp' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $paid = filter_var($criteria[ 'paid' ] ?? NULL, FILTER_VALIDATE_BOOLEAN) ? '1' : '';
+      $clientId = (int) ($criteria[ 'clientId' ] ?? 0);
+
+      $data = $this->timesheets->getAll($projectId, $kind, $dateFrom, $dateTo, $userId, $invEmpl, $invComp, $paid, $clientId, 0);
+
+      foreach ($data as $key => $row) {
+        $data[$key]['fullname'] = $this->addFullname($row);
```

Comment:
`addFullname` should be called `computeFullname` (or something similar). And I think it's superfluous:

```suggestion
        $data[$key]['fullname'] = $row['firstname'].' '.$row['lastname'];
```

Reply:
Removed

Reply:
Removed method